src/d3a/d3a_core/simulation.py

```diff
@@ -375,13 +381,15 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                 self.simulation_config.external_redis_communicator.\
                     publish_aggregator_commands_responses_events()
 
-                realtime_tick_length = time.time() - simulation_time_counter
```

Comment:
Please move all these to their own private method, since all these are related to the slowdown management.

Reply:
I still support that btw, all these could be a small private method that would handle the delays in case of either slowdown or realtime.